webui/src/pages/repositories/wizard/wizard.jsx

```diff
@@ -0,0 +1,106 @@
+import React, {useState} from "react";
+import StepWizard from "react-step-wizard";
+import {ProgressBar} from "react-bootstrap";
+import Container from "react-bootstrap/Container";
+import Row from "react-bootstrap/Row";
+import Col from "react-bootstrap/Col";
+
+const transitionDefaults = {
+    enterRight: "",
+    enterLeft: "",
+    exitRight: "",
+    exitLeft: "",
+    intro: "",
+}
+
+const Wizard = ({
+                    hashEnabled=false,
+                    transitions=transitionDefaults,
+                    progressBar = false,
+                    isGoBackButton = false,
+                    isSkipButton = false,
+                    onComplete=(any) => any,
+                    canProceed = true,
+                    onNextStep = () => {},
+                    children
+                }) => {
+
+    const [state, setState] = useState({
+        stepWizard: {},
+    });
+
+    const onStepChange = () => {
+        setState({...state});
```

Comment:
Why copy the contents of `state`, if `onNextStep` cannot access it?

Reply:
This is done to re-render the wizard controller (the buttons below that let you pass through the steps). 
If we won't do this, the `currentStep` property (and probably other ones as well) will not get updated and passed to the `WizardController` properly (via the `stepWizard={state.stepWizard}` property)

Reply:
This is yet insufficient because I need to keep the stepWizard instance somewhere where it is kept updated. 
If I use the refresh method, and in `setInstance` I assign the step wizard instance to a local variable, it will be run over when the component will re-render, thus resulting with an empty step wizard instance at any given time.
Sounds a bit confusing, I know, but we can go over it.

Reply:
I didn't mean that you should assign it to a local variable - still assign it to the state.
But here, use a separate refresh state.

Reply:
Ok cool